compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -389,6 +389,8 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
       }
       else if (name.toTermName == nme.ERROR)
         UnspecifiedErrorType
+      else if (ctx.owner.isConstructor && ctx.owner.owner.unforcedDecls.toList.exists(_.name == tree.name))
```

Comment:
I would factor out the second test into a def. It is not clear what it does. Maybe:
```scala
def isNotAccesibleFromConstructor(name: Name) = {
  val classMembers = ctx.owner.owner.unforcedDecls
  classMembers.lookup.(name).exists
}
```

Reply:
If the test is factored out it is not sufficient to ensure that it is not accessible from the constructor.

Reply:
I will use the `.lookup.(name).exists` though.